core/src/replay_stage.rs

```diff
@@ -3066,46 +3068,73 @@ impl ReplayStage {
                 fork_choice,
             );
 
+            let mut last_vote_unable_to_land = false;
+            if let Some(heaviest_bank_on_same_voted_fork) = heaviest_bank_on_same_voted_fork {
+                if let Some(last_voted_slot) = tower.last_voted_slot() {
+                    if let Some(my_latest_landed_vote) =
+                        progress.my_latest_landed_vote(heaviest_bank_on_same_voted_fork.slot())
+                    {
+                        last_vote_unable_to_land = my_latest_landed_vote < last_voted_slot
+                            && last_voted_slot < heaviest_bank_on_same_voted_fork.slot()
```

Comment:
Might be good to add a comment for this condition, is it to avoid the case where `last_voted_slot == heaviest_bank_on_same_voted_fork.slot()`?

Reply:
Yes, comment added.